pandas/_libs/window.pyx

```diff
@@ -1357,25 +1357,53 @@ cdef _roll_min_max(ndarray[numeric] input, int64_t win, int64_t minp,
     return output
 
 
+def _get_interpolation_id(str interpolation):
```

Comment:
look in ``algos.pxd`` for how we handle ``cdef enum TiebreakEnumType:``

e.g you create an enum and a dict to map to that enum. This becomes a bit simpler then.